### PR TITLE
Fixing Mixer racetest failure and enable them

### PR DESCRIPTION
**Pull request by xiaolanz**
Addresses: #3207, #3208, #3209, #3206

### CODE REVIEW

File: mixer/adapter/solarwinds/internal/appoptics/batching.go [hunk 1]
```diff
@@ -15,20 +15,27 @@
...
 	"istio.io/istio/mixer/pkg/adapter"
 )
 
+// CloseKey defines the key for synchronized map value for loopFactor.
+const CloseKey = "if-close"
+
 // BatchMeasurements reads slices of Measurement types off a channel populated by the web handler
 // and packages them into batches conforming to the limitations imposed by the API.
-func BatchMeasurements(loopFactor *bool, prepChan <-chan []*Measurement,
+func BatchMeasurements(loopFactor *sync.Map, prepChan <-chan []*Measurement,
```
Comment (geeknoid) [inline mixer/adapter/solarwinds/internal/appoptics/batching.go]: Could this be handled by just reading *loopFactor with an atomic read? Using a map as a communication mechanism seems strange.

Comment (xiaolanz) [reply mixer/adapter/solarwinds/internal/appoptics/batching.go]: I think sync.Map is idiomatic solution in go for synchronized data variables. If it's a complex structure, i would go for implementing locking pattern using mutex myself. For this particular case, using this language feature gives the cleanest fix.

Comment (mandarjog) [reply mixer/adapter/solarwinds/internal/appoptics/batching.go]: You should use sync.atom.Value.
sync.Map is not idiomatic here.

**Comment by xiaolanz** [comment]
Thanks for the suggestion. Fixed. PTAL

File: mixer/adapter/solarwinds/internal/appoptics/batching.go [hunk 2]
```diff
@@ -17,18 +17,30 @@ package appoptics
 import (
 	"time"
 
+	"sync/atomic"
```
Comment (ldemailly) [inline mixer/adapter/solarwinds/internal/appoptics/batching.go]: you typically don't want to use atomic directly, it's probably best to just use a mutex, it's rare you can solve anything with a single atomic (except low performance uuid generation)

https://golang.org/pkg/sync/atomic/

"These functions require great care to be used correctly. Except for special, low-level applications, synchronization is better done with channels or the facilities of the sync package."

**Comment by ldemailly** [comment]
thanks! can you pull master (make pull) as you are based of a master that has a bug with vendor

File: .gitmodules [hunk 3]
```diff
@@ -1,3 +0,0 @@
-[submodule "vendor"]
```
Comment (ldemailly) [inline .gitmodules]: uh no, you can't do that ! please undo that deletion

**Comment by xiaolanz** [comment]
PTAL . . I changed to use channel to avoid shared memory. and fixed the repo.

**Comment by ldemailly** [comment]
thx
/lgtm
but it’d be nice to trigger the race circleci build

**Comment by xiaolanz** [comment]
Would somebody help merge this PR? It seems being retested indefinitely.

**Comment by ldemailly** [comment]
we do need to figure out why the bot got into a loop
@sebastienvas @chxchx @hklai 

but merging this now so the flood stops

**Pull request merged**